django/contrib/gis/gdal/libgdal.py

```diff
@@ -83,23 +83,16 @@ def gdal_full_version():
     return _version_info(b'')
 
 
-version_regex = _lazy_re_compile(r'^(?P<major>\d+)\.(?P<minor>\d+)(\.(?P<subminor>\d+))?')
-
-
 def gdal_version_info():
-    ver = gdal_version().decode()
-    m = version_regex.match(ver)
+    ver = gdal_version()
+    m = re.match(br'^(?P<major>\d+)\.(?P<minor>\d+)(?:\.(?P<subminor>\d+))?', ver)
     if not m:
         raise GDALException('Could not parse GDAL version string "%s"' % ver)
-    return {key: m.group(key) for key in ('major', 'minor', 'subminor')}
+    major, minor, subminor = m.groups()
+    return (int(major), int(minor), subminor and int(subminor))
```

Comment:
This changes the value returned, but as `gdal_version_info()` is undocumented, it's fine.

Is it worth renaming to `_gdal_version_info()` as this is only used internally to populate `GDAL_VERSION`?